pkg/secrets/secrets.go

```diff
@@ -17,7 +33,52 @@ func ValidateCredentials(secret *v1.Secret) error {
 		return ValidateAWSCredentials(secret)
 	case AzureSecretType:
 		return ValidateAzureCredentials(secret)
+	case GCPSecretType:
+		return ValidateGCPCredentials(secret)
 	default:
 		return errors.Errorf("Unsupported type '%s' for secret '%s:%s'", string(secret.Type), secret.Namespace, secret.Name)
 	}
 }
+
+func getLocationType(secret *corev1.Secret) (repositoryserver.RepositoryServerSecret, error) {
+	var locationType []byte
+	var ok bool
+	if secret == nil {
+		return nil, errors.New("Secret is Nil")
+	}
+
+	if locationType, ok = (secret.Data[repositoryserver.TypeKey]); !ok {
+		return nil, errors.Errorf("secret '%s:%s' does not have required field %s", secret.Namespace, secret.Name, repositoryserver.TypeKey)
+	}
+
+	switch repositoryserver.LocType(string(locationType)) {
+	case repositoryserver.LocTypeS3:
+		return repositoryserver.NewAWSLocation(secret), nil
```

Comment:
We also need to add filesystem location type here.

Reply:
I have added empty validation function for now for filestore